src/GitHub_Updater/Rest_Update.php

```diff
@@ -248,15 +294,17 @@ public function process_request() {
 	private function get_local_branch() {
 		if ( isset( $_REQUEST['plugin'] ) ) {
 			$repos = Singleton::get_instance( 'Plugin', $this )->get_plugin_configs();
-			$repo  = $repos[ $_REQUEST['plugin'] ];
+			$repo  = isset($repos[ $_REQUEST['plugin'] ]) ? $repos[ $_REQUEST['plugin'] ] : null;
```

Comment:
This is already inside of a conditional testing this. Not sure this makes sense.